test/functional/test_framework/test_framework.py

```diff
@@ -273,6 +273,11 @@ def stop_nodes(self):
             # Wait for nodes to stop
             node.wait_until_stopped()
 
+    def restart_node(self, i):
```

Comment:
Convenient method to stop and start a node with the same arguments as before. If this gets ACK's I can push a commit to use where appropriate or submit this in a separate PR.

Reply:
Great! Ideally the stop and start methods in `TestFramework` would be methods on the `TestNode` class, and so would this, but we can clean that up later.

You can change this to:

```python
def restart_node(self, i):
    """Stop and start a test node"""
    self.stop_node(i)
    self.start_node(i)
```

As long as you also take my change to blockchain.py above.